mava/components/jax/training/trainer.py

```diff
@@ -124,4 +344,4 @@ def config_class() -> Optional[Callable]:
         Returns:
             config class/dataclass for component.
         """
-        return TrainerInitConfig
+        return CustomTrainerInitConfig
```

Comment:
I see you got a draft PR. In the tests for this, it would be important to test the custom trainer init as it requires a dictionary that conforms to unique_net_keys

Reply:
Definitely, that draft PR is currently testing the old `TrainerInit`, so once this PR goes through I'll test the custom cases separately.